src/wlapplication.cc

```diff
@@ -815,15 +817,51 @@ void WLApplication::run() {
 		if (filename_.empty()) {
 			EditorInteractive::run_editor(nullptr, EditorInteractive::Init::kDefault);
 		} else {
-			EditorInteractive::run_editor(
-			   nullptr, EditorInteractive::Init::kLoadMapDirectly, filename_, script_to_run_);
+			std::string title;
+			std::string message;
+			try {
+				if (filename_ == "last") {
+					std::optional<MapData> map = newest_edited_map();
+					if (map.has_value()) {
+						filename_ = map->filenames.at(0);
+					} else {
+						throw(WLWarning(_("No Last Edited Map"),
+						                _("Widelands could not find the last edited map.")));
+					}
+				}
+				EditorInteractive::run_editor(
+				   nullptr, EditorInteractive::Init::kLoadMapDirectly, filename_, script_to_run_);
+			} catch (const WLWarning& e) {
+				title = e.title();
+				message = e.what();
+			}
+			if (!message.empty()) {
+				g_sh->change_music(Songset::kMenu);
+				FsMenu::MainMenu m(true);
+				m.show_messagebox(title, message);
+				log_err("%s\n", message.c_str());
```

Comment:
Maybe be slightly more informative about what the game was trying to do when the error happened?

Reply:
I'm not sure what else I could report. This is only invoked when last map is requested from the command line, so there's no need for much context. And there's not much at the lower levels either that I could raise. (I used debug logging while I tried to get it to work, and considered keeping some of it as `verb_log_dbg()`, but really, there's nothing useful unless you want to spam the log with each checked path.) If there's any filesystem error along the line, it'll raise a different exception anyway.

Reply:
I've cleaned this up so that it's obvious that this message only applies when `=last` is used and lookup fails. Will that do?